device/barefoot/x86_64-accton_wedge100bf_32x-r0/plugins/eeprom.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+
+try:
+    import exceptions
+    import binascii
+    import time
+    import optparse
+    import warnings
+    import os
+    import sys
+    from sonic_eeprom import eeprom_base
+    from sonic_eeprom import eeprom_tlvinfo
+    import subprocess
+    import re
+    import pexpect
+    import cPickle as pickle
+
+except ImportError, e:
+    raise ImportError (str(e) + "- required module not found")
+
+class board(eeprom_tlvinfo.TlvInfoDecoder):
+    _TLV_INFO_MAX_LEN = 256
+    bf_commands = ["ucli", "..", "..", "bf_pltfm", "cp2112"]
+    BFCMD="do_bfshell.py -c ucli \
+                -c .. -c .. -c bf_pltfm -c cp2112 -c"
+    cache = "" 
+    DATA = {'last_poll':0, 'raw': ""}
+    CHECK_INTV = 5 #in seconds
+
+    def __init__(self, name, path, cpld_root, ro):
+        self.cache = "/tmp/.%s.cache" % os.path.basename(sys.argv[0])
+        super(board, self).__init__("", 0, '', True)
+
+    def eeprom_up2date(self):
+        if os.path.exists(self.cache):
+            self.DATA = pickle.load(open(self.cache, "rb"))
+        last_poll = self.DATA['last_poll']
+
+        #check interval
+        now = time.time()
+        if now < (last_poll + self.CHECK_INTV):
+            return True
+        else:
+            return False
+
+    def interact_with_bfshell(self, cmds, timeout=20):
+        bfshell_path = 'docker exec -it syncd /opt/bfn/install/bin/bfshell'
+        bfshell_cmd = [ bfshell_path ]
+
+        prompt_list = ['bfshell> $', 'bf-sde\.*.*> $', 'pd-switch:\d+> $', pexpect.EOF]
```

Comment:
do you you get eeprom/psu data from bfshell? Isn't your platform driver suppose create sysfs entries to expose those data?

Reply:
ok. can you get this to master branch first? we need to ensure we have this in the master branch, then we can cherry-pick it back to a release branch.

Reply:
Get it. 
I will make another PR for this change to master branch. But it needs quite a while for I'm working on other platforms.